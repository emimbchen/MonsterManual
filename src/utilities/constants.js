export const SPELL_LEVEL_COLORS = {
  0: "#B3E5FC",
  1: "#81C784",
  2: "#FFEB3B",
  3: "#FFB74D",
  4: "#F06292",
  5: "#BA68C8",
  6: "#9575CD",
  7: "#7986CB",
  8: "#64B5F6",
  9: "#4DD0E1",
};
