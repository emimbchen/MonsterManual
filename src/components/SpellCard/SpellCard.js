import { Link } from "react-router-dom";
import { SPELL_LEVEL_COLORS } from "../../utilities/constants";

function SpellCard({ spell, className = null, ...props }) {
  const { index, level, name } = spell || {};
  return (
    <div
      className={`p-3 border-l-2 drop-shadow shadow shadow-slate-50 ${className}`}
      style={{ borderLeftColor: SPELL_LEVEL_COLORS[level] }}
      {...props}
    >
      <div className="flex flex-col justify-between">
        {name && (
          <h2 className="mb-2 font-bold underline">
            <Link to={`spells/${index}`}>{name}</Link>
          </h2>
        )}

        {level && <p>Level {level}</p>}
      </div>
    </div>
  );
}

export default SpellCard;
