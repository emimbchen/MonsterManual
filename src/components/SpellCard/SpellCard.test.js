import axe from "axe-core";
import SpellCard from "./SpellCard";
import { MOCK_SPELL } from "../../utilities/mockData";
import { render, screen } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";

const mockSpellCard = (
  <MemoryRouter>
    <SpellCard spell={MOCK_SPELL} />
  </MemoryRouter>
);

describe("Spell Card Component", () => {
  test("Accessibility check", async () => {
    const { container } = render(mockSpellCard);
    const results = await axe.run(container);
    expect(results.violations.length).toBe(0);
  });

  it("renders correctly", () => {
    const { asFragment } = render(mockSpellCard);
    expect(asFragment()).toMatchSnapshot();
  });

  it("displays values", () => {
    render(mockSpellCard);
    const nameElement = screen.getByText(MOCK_SPELL.name);
    const levelElement = screen.getByText("Level " + MOCK_SPELL.level);

    expect(nameElement).toBeInTheDocument();
    expect(levelElement).toBeInTheDocument();
  });
});
