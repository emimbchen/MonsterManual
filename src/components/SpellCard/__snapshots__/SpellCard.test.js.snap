// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Spell Card Component renders correctly 1`] = `
<DocumentFragment>
  <div
    class="p-3 border-l-2 drop-shadow shadow shadow-slate-50 null"
    style="border-left-color: #FFEB3B;"
  >
    <div
      class="flex flex-col justify-between"
    >
      <h2
        class="mb-2 font-bold underline"
      >
        <a
          href="/spells/acid-arrow"
        >
          Acid Arrow
        </a>
      </h2>
      <p>
        Level 2
      </p>
    </div>
  </div>
</DocumentFragment>
`;
